jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1483,7 +1500,7 @@ public ByteBuffer getByteBuffer() throws BufferOverflowException
                         throw new BufferOverflowException();
 
                     int length = (int)size;
-                    RetainableByteBuffer combined = _pool.acquire(length, _direct);
+                    RetainableByteBuffer combined = _pool.acquire(length, _pool.isDirect());
```

Comment:
You've just added `ByteBufferPool.Sized.acquire(int size)` that could be used here.